Documentation/BootstrappingPkgsrc.md

```diff
@@ -0,0 +1,96 @@
+# Bootstrapping pkgsrc on SerenityOS
+
+## DISCLAIMER
+
+The process of bootstrapping pkgsrc stresses both the reliability and the POSIX compliance of any operating system. Due to the weaknesses of SerenityOS on both counts, this will most likely either fail, get a process stuck or crash the kernel. In fact, to date the pkgsrc bootstrap process on SerenityOS has yet to reach the finish line because of these issues.
+
+Until this works, expect endless frustration
+
+## Bootstrapping pkgsrc
+
+These steps should be executed inside the top Serenity source directory.
+1. Cross-compile SerenityOS as usual:
+```sh
+: "${ARCH:=i686}"
+./Meta/serenity.sh install ${ARCH}
+```
+2. Install all required ports:
+```sh
+for i in binutils gcc bash mawk sed grep make; do (cd Ports/$i && SERENITY_ARCH=${ARCH} ./package.sh); done
+```
+3. Clone pkgsrc:
+```sh
+git clone --single-branch --branch trunk --depth 1 https://github.com/NetBSD/pkgsrc.git Base/home/anon/pkgsrc
+```
+4. Patch `bootstrap/bootstrap`:
+```sh
+cat >Base/home/anon/pkgsrc-bootstrap.patch <<EOF
+diff a/bootstrap/bootstrap b/bootstrap/bootstrap
+--- a/bootstrap/bootstrap
++++ b/bootstrap/bootstrap
+@@ -805,6 +805,13 @@
+ SCO_SV)
+ 	# /bin/sh under OpenServer 5.0.7/3.2 breaks bmake tests.
+ 	#bmakexargs="$bmakexargs --with-defshell=/bin/ksh"
+ 	;;
++SerenityOS)
++	root_group=root
++	machine_arch=`uname -m`
++	case "$machine_arch" in
++	i?86) machine_arch="i386" ;;
++	esac
++	;;
+ SunOS)
+ 	root_group=root
+ 	need_bsd_install=yes
+EOF
+patch -d Base/home/anon/pkgsrc/ -p1 <Base/home/anon/pkgsrc-bootstrap.patch
```

Comment:
Can this be placed somewhere inside `Meta/` so we don't have to bother with manually creating the file?
In fact, why is this whole procedure not in `Ports/`?

Reply:
Because pkgsrc isn't a port in the `Ports/` sense, it doesn't make sense to put it there. These patches should instead be merged upstream so that they are no longer needed.

This is still highly experimental, so a page of "copy-paste this into a shell" to reproduce is probably the best approach for now.